File: PseudoCoupHQ/Research/oracle/riscv/softfloat_slices/arch_units/cpp/au_213_c_alignof_gnu_f64.cpp
// arch-unit 213  --  c  `__alignof a`  lhs=double rhs=None
// symbol op_64   outcome LIFTED   2 arch-opcodes
//
// the arch-unit, arch-opcode by arch-opcode:
//   c.li a0, 0x8                         integer    constant
//   c.jr ra                              integer    return
//
// answer: uint64_t, 64 bits.  parameters are operand bit patterns.
#include "arch_units_int.hpp"

namespace archunits {

uint64_t au_213_c_alignof_gnu_f64(uint64_t p0)
{
    /* fa0: operand `a` (double) arrives in fa0 as a bit pattern */
    const uint64_t v1 = p0;
    const uint64_t v2 = UINT64_C(0x8);
    /* the answer is uint64_t, 64 bits */
    return v2;
}

}  // namespace archunits

File: PseudoCoupHQ/Research/oracle/riscv/softfloat_slices/arch_units/rust/au_213_c_alignof_gnu_f64.rs
// arch-unit 213  --  c  `__alignof a`  lhs=double rhs=None
// symbol op_64   outcome LIFTED   2 arch-opcodes
//
// the arch-unit, arch-opcode by arch-opcode:
//   c.li a0, 0x8                         integer    constant
//   c.jr ra                              integer    return
//
// answer: uint64_t, 64 bits.  parameters are operand bit patterns.
#![allow(unused_parens, unused_variables, unused_imports, clippy::all)]
use crate::au_int::*;

pub fn au_213_c_alignof_gnu_f64(p0: u64) -> u64 {
    // fa0: operand `a` (double) arrives in fa0 as a bit pattern
    let v1: u64 = p0;
    let v2: u64 = 0x8u64;
    // the answer is uint64_t, 64 bits
    v2
}
